src/main/scala/com/microsoft/sqlserver/jdbc/spark/utils/BulkCopyUtils.scala

```diff
@@ -239,45 +261,71 @@ object BulkCopyUtils extends Logging {
         val dfCols = df.schema
 
         val tableCols = getSchema(rs, JdbcDialects.get(url))
+        val ccol = getComputedCols(conn, options.dbtable)
+        var ccolList : List[String] = List()
+        while (ccol.next()) {
+            val ccolName = ccol.getString("name")
+            ccolList = ccolList :+ ccolName
+        }
+
         val prefix = "Spark Dataframe and SQL Server table have differing"
 
-        assertCondition(dfCols.length == tableCols.length,
-            s"${prefix} numbers of columns")
+        if (ccolList.length == 0) {
+            assertCondition(dfCols.length == tableCols.length,
+                s"${prefix} numbers of columns")
+        }
+        else {
+            val dfColNames =  df.schema.fieldNames.toList
+            var ccolCount = 0
+            for (j <- 0 to ccolList.length-1){
+                if (isCaseSensitive && dfColNames.contains(ccolList(j)) ||
+                  !isCaseSensitive && dfColCaseMap.contains(ccolList(j).toLowerCase()) && dfColCaseMap(ccolList(j).toLowerCase()) == ccolList(j)) {
+                    ccolCount += 1
+                }
+            }
+            assertCondition(dfCols.length-ccolCount+ccolList.length == tableCols.length,
+                s"${prefix} numbers of columns")
+        }
 
         val result = new Array[ColumnMetadata](tableCols.length)
 
         for (i <- 0 to tableCols.length-1) {
             val tableColName = tableCols(i).name
             var dfFieldIndex = 0
-            var dfColName:String = ""
-            if (isCaseSensitive) {
-                dfFieldIndex = dfCols.fieldIndex(tableColName)
-                dfColName = dfCols(dfFieldIndex).name
-                assertCondition(tableColName == dfColName,
-                    s"""${prefix} column names '${tableColName}' and
-                        '${dfColName}' at column index ${i} (case sensitive)""")
-            } else {
-                dfFieldIndex = dfCols.fieldIndex(dfColCaseMap(tableColName.toLowerCase()))
-                dfColName = dfCols(dfFieldIndex).name
-                assertCondition(tableColName.toLowerCase() == dfColName.toLowerCase(),
-                    s"""${prefix} column names '${tableColName}' and
-                        '${dfColName}' at column index ${i} (case insensitive)""")
-            }
+            if (ccolList.contains(tableColName)) {
+                dfFieldIndex = -1
```

Comment:
A comment would help. This looks like a hack to set dfFieldIndex = -1. Ideally it would be good to have a specific field say isComputed; but that's an extra variable added to dfColMetaData. IMO we should use isAutoIncrement

Reply:
Now using isAutoIncrement